pkg/model/message.go

```diff
@@ -14,6 +15,64 @@ type Messages struct {
 	Original bool
 }
 
+type MessagesSlice []Messages
+
+// HasLanguage checks if MessagesSlice contains Messages with the given language.
+func (ms MessagesSlice) HasLanguage(targetLang language.Tag) bool {
+	return slices.ContainsFunc(ms, func(m Messages) bool {
+		return m.Language == targetLang
+	})
+}
+
+// LanguageIndex returns index of Messages with the given language. If not found, returns -1.
+func (ms MessagesSlice) LanguageIndex(targetLang language.Tag) int {
```

Comment:
Always try to use single-word variable names, and add two or three words if they add meaningful context and meaning.

```suggestion
func (ms MessagesSlice) LanguageIndex(lang language.Tag) int {
```